ui/raidboss/data/06-ew/trial/rubicante.ts

```diff
@@ -66,16 +84,79 @@ const triggerSet: TriggerSet<Data> = {
       netRegex: { id: '7CEA', source: 'Rubicante', capture: false },
       response: Responses.aoe(),
     },
+    {
+      id: 'Rubicante Ordeal of Purgation Line',
+      type: 'MapEffect',
+      netRegex: { flags: purgationLineFlags, location: purgationLocations },
+      run: (data, matches) => {
+        data.purgationLoc = matches.location;
+        data.purgationLine = matches.flags;
+      },
+    },
+    {
+      id: 'Rubicante Ordeal of Purgation Rotate',
+      type: 'MapEffect',
+      netRegex: { flags: purgationMidRotateFlags, location: purgationMidRingLoc },
+      run: (data, matches) => data.purgationMidRotate = matches.flags,
+    },
     {
       id: 'Rubicante Ordeal of Purgation',
       type: 'StartsUsing',
       netRegex: { id: ['7CC4', '80E8'], source: 'Rubicante', capture: false },
+      delaySeconds: 0.2,
       durationSeconds: 8,
-      alertText: (_data, _matches, output) => output.avoid!(),
+      alertText: (data, _matches, output) => {
+        let idx = data.purgationLoc ? purgationLocations.indexOf(data.purgationLoc) : undefined;
+        if (idx === undefined || idx === -1)
+          return output.avoidCone!();
+
+        // adjust idx for 2nd & 3rd+ purgations based on jagged middle lines or rotation
+        if (data.purgation === 2) {
+          // 2nd purgation - CW/CCW line, no rotation
+          const jaggedLineFlags = purgationLineFlags.slice(1);
```

Comment:
Can you add a comment about what this slice is doing? I'm not sure I understand it.